interop-testing/src/main/java/io/grpc/testing/integration/AbstractInteropTest.java

```diff
@@ -2006,34 +1997,60 @@ public void performSoakTest(
       int soakIterations,
       int maxFailures,
       int maxAcceptablePerIterationLatencyMs,
+      int minTimeMsBetweenRpcs,
       int overallTimeoutSeconds)
       throws Exception {
     int iterationsDone = 0;
     int totalFailures = 0;
     Histogram latencies = new Histogram(4 /* number of significant value digits */);
     long startNs = System.nanoTime();
+    ManagedChannel soakChannel = createChannel();
+    TestServiceGrpc.TestServiceBlockingStub soakStub = TestServiceGrpc
+        .newBlockingStub(soakChannel)
+        .withInterceptors(recordClientCallInterceptor(clientCallCapture));
+    soakStub = soakStub.withInterceptors(recordClientCallInterceptor(clientCallCapture));
     for (int i = 0; i < soakIterations; i++) {
       if (System.nanoTime() - startNs >= TimeUnit.SECONDS.toNanos(overallTimeoutSeconds)) {
         break;
       }
-      SoakIterationResult result = performOneSoakIteration(resetChannelPerIteration);
+      long earliestNextStartNs = System.nanoTime()
+          + TimeUnit.MILLISECONDS.toNanos(minTimeMsBetweenRpcs);
+      if (resetChannelPerIteration) {
+        soakChannel.shutdownNow();
+        soakChannel.awaitTermination(10, TimeUnit.SECONDS);
+        soakChannel = createChannel();
+        soakStub = TestServiceGrpc
+            .newBlockingStub(soakChannel)
+            .withInterceptors(recordClientCallInterceptor(clientCallCapture));
+      }
+      SoakIterationResult result = performOneSoakIteration(soakStub);
+      String peer = clientCallCapture
+          .get().getAttributes().get(Grpc.TRANSPORT_ATTR_REMOTE_ADDR).toString();
       System.err.print(
           String.format(
-              "soak iteration: %d elapsed: %d ms", i, result.getLatencyMs()));
+              "soak iteration: %d elapsed_ms: %d peer: %s",
+              i, result.getLatencyMs(), peer));
       if (!result.getStatus().equals(Status.OK)) {
         totalFailures++;
         System.err.println(String.format(" failed: %s", result.getStatus()));
       } else if (result.getLatencyMs() > maxAcceptablePerIterationLatencyMs) {
         totalFailures++;
         System.err.println(
             String.format(
-                " exceeds max acceptable latency: %d", maxAcceptablePerIterationLatencyMs));
+                " exceeds max acceptable latency: %d",
+                maxAcceptablePerIterationLatencyMs));
       } else {
         System.err.println(" succeeded");
       }
       iterationsDone++;
       latencies.recordValue(result.getLatencyMs());
+      long remainingNs = earliestNextStartNs - System.nanoTime();
+      if (remainingNs > 0) {
+        Thread.sleep(TimeUnit.MILLISECONDS.convert(remainingNs, TimeUnit.NANOSECONDS));
```

Comment:
Protip: TimeUnit has sleep(). Pretty strange it is there, really, but `TimeUnit.NANOSECONDS.sleep(remainingNs)` does at least read well.

No need to change this, but wanted to mention it.

Reply:
done